pkg/lib/flavor/util/platform_flavor_util.go

```diff
@@ -124,44 +125,44 @@ func (pfutil PlatformFlavorUtil) GetMetaSectionDetails(hostDetails *taModel.Host
 		meta.Schema = pfutil.getSchema()
 
 	case common.FlavorPartAssetTag:
-		description.FlavorPart = flavorPartName.String()
+		description[cm.FlavorPart] = flavorPartName.String()
 		if hostDetails != nil {
 			hwuuid, err := uuid.Parse(hostDetails.HardwareUUID)
 			if err != nil {
 				return nil, errors.Wrapf(err, "Invalid Hardware UUID for %s FlavorPart", flavorPartName)
 			}
-			description.HardwareUUID = &hwuuid
+			description[cm.HardwareUUID] = hwuuid.String()
 
 			if hostDetails.HostName != "" {
-				description.Source = strings.TrimSpace(hostDetails.HostName)
+				description[cm.Source] = strings.TrimSpace(hostDetails.HostName)
 			}
 		} else if tagCertificate != nil {
 			hwuuid, err := uuid.Parse(tagCertificate.Subject)
 			if err != nil {
 				return nil, errors.Wrapf(err, "Invalid Hardware UUID for %s FlavorPart", flavorPartName)
 			} else {
-				description.HardwareUUID = &hwuuid
+				description[cm.HardwareUUID] = hwuuid.String()
 			}
 		}
-		description.Label = pfutil.getLabelFromDetails(meta.Vendor.String(), (*description.HardwareUUID).String(), pfutil.getCurrentTimeStamp())
+		description[cm.Label] = pfutil.getLabelFromDetails(meta.Vendor.String(), description[cm.HardwareUUID].(string), pfutil.getCurrentTimeStamp())
 
 	case common.FlavorPartHostUnique:
 		if hostDetails != nil {
 			if hostDetails.HostName != "" {
-				description.Source = strings.TrimSpace(hostDetails.HostName)
+				description[cm.Source] = strings.TrimSpace(hostDetails.HostName)
 			}
 			hwuuid, err := uuid.Parse(hostDetails.HardwareUUID)
 			if err != nil {
 				return nil, errors.Wrapf(err, "Invalid Hardware UUID for %s FlavorPart", flavorPartName)
 			}
-			description.HardwareUUID = &hwuuid
+			description[cm.HardwareUUID] = hwuuid.String()
 		}
-		description.BiosName = biosName
-		description.BiosVersion = biosVersion
-		description.OsName = osName
-		description.OsVersion = osVersion
-		description.FlavorPart = flavorPartName.String()
-		description.Label = pfutil.getLabelFromDetails(meta.Vendor.String(), (*description.HardwareUUID).String(), pfutil.getCurrentTimeStamp())
+		description[cm.BiosName] = biosName
+		description[cm.BiosVersion] = biosVersion
+		description[cm.OsName] = osName
+		description[cm.OsVersion] = osVersion
+		description[cm.FlavorPart] = flavorPartName.String()
+		description[cm.Label] = pfutil.getLabelFromDetails(meta.Vendor.String(), description[cm.HardwareUUID].(string), pfutil.getCurrentTimeStamp())
```

Comment:
try to use a consistent variable name- some places it is model, fm and here it is cm